serial_windows.go

```diff
@@ -19,46 +19,29 @@ package serial
 
 import (
 	"sync"
-	"syscall"
 	"time"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/registry"
 )
 
 type windowsPort struct {
 	mu     sync.Mutex
-	handle syscall.Handle
+	handle windows.Handle
 }
 
 func nativeGetPortsList() ([]string, error) {
-	subKey, err := syscall.UTF16PtrFromString("HARDWARE\\DEVICEMAP\\SERIALCOMM\\")
+	key, err := registry.OpenKey(windows.HKEY_LOCAL_MACHINE, `HARDWARE\DEVICEMAP\SERIALCOMM\`, windows.KEY_READ)
 	if err != nil {
-		return nil, &PortError{code: ErrorEnumeratingPorts}
-	}
-
-	var h syscall.Handle
-	if err := syscall.RegOpenKeyEx(syscall.HKEY_LOCAL_MACHINE, subKey, 0, syscall.KEY_READ, &h); err != nil {
-		if errno, isErrno := err.(syscall.Errno); isErrno && errno == syscall.ERROR_FILE_NOT_FOUND {
-			return []string{}, nil
-		}
-		return nil, &PortError{code: ErrorEnumeratingPorts}
```

Comment:
I recall this check has been added because sometimes the enumerator would "fail" to get ports when there were no ports available: https://github.com/bugst/go-serial/pull/59

Is it still possible to intercept the `ERROR_FILE_NOT_FOUND`?
Maybe we could check if the assertion works by opening a non-existing key.

Reply:
Done.